### PR TITLE
Fix error creating and getting node in OpenStack

### LINKED ISSUE #1672: Error creating and getting node in OpenStack when setting ex_force_microversion
## Summary

An error is reported when creating node in OpenStack:
`KeyError: 'id'`

**When `ex_force_microversion` is set to a version newer than 2.47.**

## Detailed Information

It seems that there is an error with the api_node about `flavor` information.

```py
    node = driver.create_node(**args)
  File "/usr/local/lib/python3.8/dist-packages/libcloud/compute/drivers/openstack.py", line 1569, in create_node
    return self._to_node(server_object)
  File "/usr/local/lib/python3.8/dist-packages/libcloud/compute/drivers/openstack.py", line 2523, in _to_node
    flavorId=api_node["flavor"]["id"],
```

There are changes in OpenStack API:

_Before microversion 2.47 this contains the ID and links for the flavor used to boot the server instance. This can be an empty object in case flavor information is no longer present in the system._

_As of microversion 2.47 this contains a subset of the actual flavor information used to create the server instance, represented as a nested dictionary._

See details:
https://docs.openstack.org/api-ref/compute/?expanded=create-server-detail,show-server-details-detail#show-server-details

So it seems that as of microversion 2.47 the field `flavor.id` is not returned by the API call.

### PR DESCRIPTION
**Pull request by micafer**
### Description

Fixes #1672

### Status

Replace this: describe the PR status. Examples:

- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [x] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by micafer** [comment]
I also update the black version to avoid this error:
https://github.com/psf/black/issues/2964

**Comment by Kami** [comment]
LGTM.

Is this bug related to #1648? Aka would it also happen if ``ex_force_microversion`` argument is not set or not?

**Review by Kami — APPROVED** [review]

**Comment by micafer** [comment]
> Is this bug related to #1648? Aka would it also happen if `ex_force_microversion` argument is not set or not?

Yes. It is related. It only fails if ex_force_microversion is set to a version newer than 2.47.

File: CHANGES.rst [hunk 1]
```diff
@@ -23,6 +23,10 @@ Compute
   (GITHUB-1661, GITHUB-1661)
   [Dimitris Galanis - @dimgal1]
 
+- [OpenStack] Fix error creating and getting node in OpenStack.
```
Comment (Kami) [inline CHANGES.rst]: Please add some more details aka that it only affected scenarios where ``ex_force_microversion`` was set to ``> 2.47``.

Comment (micafer) [reply CHANGES.rst]: Updated.

**Pull request merged**